### PR TITLE
LPD-41688 Improving Headless Test Suits

### PR DESCRIPTION
**Pull request by ZakiMiklos**
https://liferay.atlassian.net/browse/LPD-41487

### CODE REVIEW

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf ci:test:help

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
@ZakiMiklos have you tested the new test suites somewhere?

**Comment by ZakiMiklos** [comment]
ci:help

**Comment by ZakiMiklos** [comment]
ci:test:headless-poshi

**Comment by ZakiMiklos** [comment]
ci:test:headless-poshi

**Comment by ZakiMiklos** [comment]
ci:help

**Comment by ZakiMiklos** [comment]
ci:help

**Comment by ZakiMiklos** [comment]
ci:help

**Comment by ZakiMiklos** [comment]
ci:test:headless-poshi

**Comment by ZakiMiklos** [comment]
ci:help

File: test.properties [hunk 1]
```diff
@@ -5759,10 +5711,57 @@
...
+        ) AND \
+        (portal.acceptance == true) AND \
+        (portal.upstream == true) AND \
+        (testray.main.component.name == "Headless API Builder")
+
+    #
+    # Headless Functional (Playwright + Poshi)
+    #
+
+    playwright.test.project[playwright-js-tomcat90-mysql57][headless-functional]=\
```
Comment (magjed4289) [inline test.properties:5759]: I think you can use ${headless.playwright.projects} instad of listing the files

**Comment by ZakiMiklos** [comment]
ci:test:headless-functional

**Comment by magjed4289** [comment]
ci:test:headless-poshi

**Comment by ccorreagg** [comment]
Any update here?

**Comment by magjed4289** [comment]
ci:test:headless-poshi:dryrun

**Comment by magjed4289** [comment]
headless-object-entry-environments-integration:dryrun

**Comment by magjed4289** [comment]
headless-object-entry-environments-integration:dryrun

**Comment by magjed4289** [comment]
ci:help

**Comment by magjed4289** [comment]
ci:test:headless-acceptance:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-api-builder:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-api-builder-functional:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-functional:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-object-entry-db-integration:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-object-entry-environments-integration:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-playwright:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-staging:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-staging-custom-db:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-poshi:dryrun

**Comment by magjed4289** [comment]
@ZakiMiklos could you please review:

- headless-api-builder - no functional tests running, see the batch summaries [here](https://test-1-37.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/19136/job-summary/index.html)

- headless-object-entry-environments-integration - only functional-upgrade-weblogic141-mysql57 runs, please see [here](https://test-1-31.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11774/job-summary/index.html)

- headless-staging-custom-db - nothing runs, please see [here](https://test-1-33.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/7124/job-summary/index.html)

After each change you can run each suite with `:dryrun` param, it will generate new Job Summary

**Comment by ZakiMiklos** [comment]
ci:test:sf

**Comment by ZakiMiklos** [comment]
ci:test:headless-api-builder:dryrun

**Comment by ZakiMiklos** [comment]
ci:test:headless-object-entry-environments-integration:dryrun

**Comment by ZakiMiklos** [comment]
Made some changes @magjed4289 

I don't know why the headless-staging-custom-db failing that part of property was unchanged

**Comment by magjed4289** [comment]
ci:help

**Comment by magjed4289** [comment]
@ZakiMiklos I checked, it is correct behaviour without setting the db previously https://liferay.atlassian.net/wiki/spaces/ENGSTAGING/pages/2324693263/How+to+run+Staging+test+cases+on+a+customer+s+DB+on+Jenkins

**Comment by magjed4289** [comment]
@ZakiMiklos in the task description we have:
- Create ci:test:headless-object-entry-db-integration - Test Integration Object Entry REST API tests with all the databases.

- Create ci:test:headless-object-entry-environments-integration - Test Integration Object Entry REST API tests with all the available environments.
I understand in these two suites the integration tests should run, so the property `test.batch.names` should be set accordingly and there's no need to define any properties for functional tests

**Comment by magjed4289** [comment]
ci:test:headless-object-entry-environments-integration:dryrun

**Comment by magjed4289** [comment]
@ZakiMiklos with changes from [this commit ](https://github.com/magjed4289/liferay-portal/pull/1340/commits/a28ac8249971d84bea1bfa60373d2d2801711e53) the routine 
headless-object-entry-environments-integration seems to work as expected. Could you please include it in your PR and double check? With this change I think all the rest is working correctly. Thanks! 🙏🏻

**Comment by magjed4289** [comment]
ci:test:headless-object-entry-environments-integration:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-object-entry-db-integration:dryrun

**Comment by magjed4289** [comment]
ci:headless-object-entry-environments-integration:dryrun

**Comment by magjed4289** [comment]
ci:test:headless-object-entry-environments-integration:dryrun

**Comment by magjed4289** [comment]
ci:forward

**Pull request closed without merging**